diff -upr lib7zip-1.6.5.org/configure.ac lib7zip-1.6.5/configure.ac
--- lib7zip-1.6.5.org/configure.ac	2013-04-13 16:56:39.000000000 +0200
+++ lib7zip-1.6.5/configure.ac	2015-09-27 17:42:27.000000000 +0200
@@ -1,6 +1,6 @@
 AC_INIT([lib7zip], [1.6.5], [jingnan.si@gmail.com])
 AC_CONFIG_SRCDIR(Lib7Zip/lib7zip.h)
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_CANONICAL_HOST
@@ -12,7 +12,8 @@ AC_C_INLINE
 AC_HEADER_STDBOOL
 
 AC_PROG_RANLIB
-LT_INIT([dlopen win32-dll])
+AM_PROG_AR
+#LT_INIT([dlopen win32-dll])
 
 AC_LANG(C++)
 
@@ -60,7 +61,6 @@ AC_CONFIG_FILES([
 	Makefile
 	includes/Makefile
 	Lib7Zip/Makefile
-	Test7Zip/Makefile
 ])
 
 AC_OUTPUT
diff -upr lib7zip-1.6.5.org/Lib7Zip/Makefile.am lib7zip-1.6.5/Lib7Zip/Makefile.am
--- lib7zip-1.6.5.org/Lib7Zip/Makefile.am	2013-04-13 16:56:39.000000000 +0200
+++ lib7zip-1.6.5/Lib7Zip/Makefile.am	2015-09-21 03:05:11.000000000 +0200
@@ -3,9 +3,9 @@ noinst_LIBRARIES = lib7zip.a
 if WIN32
 A_EXTRA_FLAGS = -DUNICODE -D_UNICODE
 A_LIBS = -loleaut32 -luuid
-else
-A_EXTRA_FLAGS = 
-A_LIBS = -no-undefined
+#else
+#A_EXTRA_FLAGS = 
+#A_LIBS = -no-undefined
 endif
 
 AM_CPPFLAGS = -I../includes -I../includes/CPP -I../includes/CPP/myWindows -I../includes/CPP/include_windows \
@@ -38,29 +38,27 @@ SOURCES = \
 NODIST_SOURCES = ../includes/CPP/Common/MyWindows.cpp ../includes/CPP/Common/StdAfx.h \
 	../includes/CPP/Windows/PropVariant.cpp ../inclues/CPP/Windows/StdAfx.h
 
-lib7zip_a_SOURCES = $(SOURCES)
-
-nodist_lib7zip_a_SOURCES= $(NODIST_SOURCES)
+lib7zip_a_SOURCES = $(SOURCES) $(NODIST_SOURCES)
 
 lib7zip_a_CFLAGS = $(AM_CFLAGS)
 lib7zip_a_CPPFLAGS = $(AM_CPPFLAGS)
 
-if WIN32
-lib7zip.dll:
-	$(CXXLD) -shared $(AM_CXXFLAGS) \
-	$(CXXFLAGS) $(lib7zip_a_LDFLAGS) $(LDFLAGS) -o $@ $(lib7zip_a_OBJECTS) $(A_LIBS)
-
-all-local:lib7zip.dll
-	-cp lib7zip.dll ../lib
+#if WIN32
+#lib7zip.dll:
+#	$(CXXLD) -shared $(AM_CXXFLAGS) \
+#	$(CXXFLAGS) $(lib7zip_a_LDFLAGS) $(LDFLAGS) -o $@ $(lib7zip_a_OBJECTS) $(A_LIBS)
+
+#all-local:lib7zip.dll
+#	-cp lib7zip.dll ../lib
+
+#clean-local:
+#	-rm -f lib7zip.dll
+
+#else
+#lib_LTLIBRARIES = lib7zip.la
+#lib7zip_la_SOURCES = $(SOURCES)
 
-clean-local:
-	-rm -f lib7zip.dll
-
-else
-lib_LTLIBRARIES = lib7zip.la
-lib7zip_la_SOURCES = $(SOURCES)
-
-nodist_lib7zip_la_SOURCES= $(NODIST_SOURCES)
+#nodist_lib7zip_la_SOURCES= $(NODIST_SOURCES)
 
 # The -no-undefined makes it possible to build DLLs for Windows,
 # or shared libraries for Tru64 or AIX (according to the autobook
@@ -69,8 +67,8 @@ nodist_lib7zip_la_SOURCES= $(NODIST_SOUR
 # -export-symbols lib7zip.sym \
 #LIB7ZIP_VERSION_INFO=1.6.0
 # -version-info @LIB7ZIP_VERSION_INFO@
-lib7zip_la_LDFLAGS = $(A_LIBS)
-endif
+#lib7zip_la_LDFLAGS = $(A_LIBS)
+#endif
 
 EXTRA_DIST = 
 
diff -upr lib7zip-1.6.5.org/Makefile.am lib7zip-1.6.5/Makefile.am
--- lib7zip-1.6.5.org/Makefile.am	2012-07-30 16:24:58.000000000 +0200
+++ lib7zip-1.6.5/Makefile.am	2015-09-21 03:03:57.000000000 +0200
@@ -1,5 +1,5 @@
 ACLOCAL_AMFLAGS=-I m4
 
-SUBDIRS = Lib7Zip includes Test7Zip
+SUBDIRS = Lib7Zip includes
 
 EXTRA_DIST = autogen.sh
diff -upr 7z922.org/CPP/7zip/MyVersion.h 7z922/CPP/7zip/MyVersion.h
--- 7z922.org/CPP/7zip/MyVersion.h	2011-04-18 10:05:28.000000000 +0200
+++ 7z922/CPP/7zip/MyVersion.h	2015-09-21 01:54:33.000000000 +0200
@@ -1,4 +1,4 @@
-#include "..\..\C\7zVersion.h"
+#include "../../C/7zVersion.h"
 
 #undef MY_COPYRIGHT
 #undef MY_VERSION_COPYRIGHT_DATE
