--- p7zip_9.20.1/CPP/myWindows/StdAfx.h.org	2018-07-30 19:55:30.551292335 +0200
+++ p7zip_9.20.1/CPP/myWindows/StdAfx.h	2018-07-30 19:59:07.339297487 +0200
@@ -52,7 +52,9 @@ EXTERN_C_END
 
 /***************************/
 
+#ifndef CLASS_E_CLASSNOTAVAILABLE
 #define CLASS_E_CLASSNOTAVAILABLE        ((HRESULT)0x80040111L)
+#endif
 
 /************************* LastError *************************/
 inline DWORD WINAPI GetLastError(void) { return errno; }
