diff -upr xdelta-3.1.0.org/configure.ac xdelta-3.1.0/configure.ac
--- xdelta-3.1.0.org/configure.ac	2016-01-08 08:58:54.000000000 +0100
+++ xdelta-3.1.0/configure.ac	2018-12-24 00:41:54.216409998 +0100
@@ -20,7 +20,7 @@ AC_ARG_WITH(
         [--with-liblzma],
         [build with liblzma support @<:@default=autodetect@:>@])],
     [USE_LIBLZMA=$withval],
-    [USE_LIBLZMA=auto])
+    [USE_LIBLZMA=no])
 
 if test "x$USE_LIBLZMA" != xno ; then
     AC_CHECK_HEADERS([lzma.h],,[
diff -upr xdelta-3.1.0.org/xdelta3.h xdelta-3.1.0/xdelta3.h
--- xdelta-3.1.0.org/xdelta3.h	2016-01-08 08:58:54.000000000 +0100
+++ xdelta-3.1.0/xdelta3.h	2018-12-24 00:42:21.179128135 +0100
@@ -28,9 +28,9 @@
 #define _ISOC99_SOURCE
 #define _C99_SOURCE
 
-#if HAVE_CONFIG_H
-#include "config.h"
-#endif
+//#if HAVE_CONFIG_H
+#include "../extra/xdelta-3.1.0/config.h"
+//#endif
 
 #include <errno.h>
 #include <stdarg.h>
diff -upr xdelta-3.1.0.org/Makefile.am xdelta-3.1.0/Makefile.am
--- xdelta-3.1.0.org/Makefile.am	2019-03-18 14:33:02.205771843 +0100
+++ xdelta-3.1.0/Makefile.am	2019-03-18 14:32:01.627336297 +0100
@@ -4,7 +4,7 @@ AUTOMAKE_OPTIONS = subdir-objects
 bin_PROGRAMS = xdelta3
 noinst_PROGRAMS = xdelta3regtest xdelta3decode xdelta3checksum
 
-export AFL_HARDEN
+#export AFL_HARDEN
 
 common_SOURCES = \
 	  xdelta3-blkcache.h \
